plugins/ModuleLevelTrigger.cpp

```diff
@@ -509,12 +509,12 @@ ModuleLevelTrigger::add_tc(const triggeralgs::TriggerCandidate& tc)
 
     for (std::vector<PendingTD>::iterator it = m_pending_tds.begin(); it != m_pending_tds.end();) {
       if (check_overlap(tc, *it)) {
-        if (m_use_readout_map) {
+	it->contributing_tcs.push_back(tc);
+        if ( (m_use_readout_map) && (m_readout_window_map.count(tc.type)) ){
```

Comment:
I cannot select this all, but lines 511 - 536, that's a very, nested if statement,  makes it difficult to read and understand... normally something like this:

```
for(i = 0; i < n; ++i){
   if(this_true){
      if(that_als_true){
         50 lines of code
      }
   }
}
```

can be easily changed withs simple guard clauses:

```
for(i = 0; i < n; ++i){
   if(!this_true)
      continue;
   if(!that_also_true)
      continue;

   50 lines of code
}
```

Can read more about them on the refactoring guru page: https://refactoring.guru/replace-nested-conditional-with-guard-clauses

It's a comment though, PR still approved :+1: